src/infiniteLoops/__tests__/runtime.ts

```diff
@@ -64,13 +68,22 @@ test('works in runInContext when throwInfiniteLoops is false', async () => {
   expect(result?.[1]).toBe(false)
 })
 
+const testForInfiniteLoopWithCode = (code: string, previousPrograms: es.Program[]) => {
+  const context = createContext(Chapter.SOURCE_4, Variant.DEFAULT)
+  const program = parse(code, context)
+  if (program === undefined) {
+    return
```

Comment:
Given that `testForInfiniteLoop` can return `undefined`, it would be better to throw an error here, else there's a possibility that testcases with `undefined` as the expected result will pass even if the code fails the parse.